Fw/Buffer/Buffer.hpp

```diff
@@ -93,7 +93,18 @@ class Buffer : public Fw::Serializable {
     //! to the wrapped buffer. Once obtained the user should call one of two functions: `sbb.resetSer();` to setup for
     //! serialization, or `sbb.setBuffLen(buffer.getSize());` to setup for deserializing.
     //! \return representation of the wrapped data to aid in serializing to it
-    SerializeBufferBase& getSerializeRepr();
+    DEPRECATED(SerializeBufferBase& getSerializeRepr(), "Switch to .getSerializer() and .getDeserializer()");
+
+    //! Returns a ExternalSerializeBufferWithMemberCopy representation of the wrapped data for serializing
+    //!
+    //! \return representation of the wrapped data to aid in serializing to it
+    ExternalSerializeBufferWithMemberCopy getSerializer();
+
+
+    //! Returns a ExternalSerializeBufferWithMemberCopy representation of the wrapped data for deserializing
+    //!
```

Comment:
Should add a `\warning` say that this assumes the entire buffer (up to `getSize`) is available for deserialization.